src/model/event.rs

```diff
@@ -473,23 +477,22 @@ impl CacheUpdate for GuildMemberUpdateEvent {
         cache.update_user_entry(&self.user).await;
 
         if let Some(guild) = cache.guilds.write().await.get_mut(&self.guild_id) {
-            let mut found = false;
-
             let item = if let Some(member) = guild.members.get_mut(&self.user.id) {
                 let item = Some(member.clone());
 
+                member.joined_at.clone_from(&Some(self.joined_at));
                 member.nick.clone_from(&self.nick);
                 member.roles.clone_from(&self.roles);
                 member.user.clone_from(&self.user);
-
-                found = true;
+                member.pending.clone_from(&self.pending);
+                member.premium_since.clone_from(&self.premium_since);
 
                 item
             } else {
                 None
             };
 
-            if !found {
+            if item.is_none() {
                 guild.members.insert(self.user.id, Member {
```

Comment:
Should this be inserting `Some(self.joined_at)` in addition to the other `pending` and `premium_since` fields on `self` instead of None/false values?

Reply:
It should, good catch! I made this change without much thought, but taking a second look, the `Member` instance should contain data from the update event when it's missing.